git/git.go

```diff
@@ -26,6 +27,51 @@ func HeadCommit(path string) (string, error) {
 	return strings.Trim(string(output), "\n"), err
 }
 
+func formatGitURL(url, branch string) string {
+	splitURL := strings.Split(url, "/")
+
+	if strings.HasPrefix(url, "https://github.com") && len(splitURL) > 4 {
+		org := splitURL[3]
+		repo := strings.TrimSuffix(splitURL[4], ".git")
+
+		return fmt.Sprintf("https://api.github.com/repos/%s/%s/commits/%s", org, repo, branch)
+	} else if strings.HasPrefix(url, "https://git.rancher.io") && len(splitURL) > 3 {
+		org := "rancher"
+		repo := strings.TrimSuffix(splitURL[3], ".git")
+
+		return fmt.Sprintf("https://api.github.com/repos/%s/%s/commits/%s", org, repo, branch)
```

Comment:
This needs to go to an endpoint we create on the git mirror that works the same as GitHub's.  The main point of git.rancher.io is to be an endpoint people can whitelist (or not have blacklisted in the first place) that *isn't* the entirety of the GitHub world.